### PR TITLE
SOAP envelope 1.2 schema valid Fault response

### PR DESCRIPTION
**Pull request by sigitas-stumbras**

### CODE REVIEW

**Comment by jsdevel** [comment]
please fix the build and keep coverage up with tests.

**Comment by sigitas-stumbras** [comment]
Is this good enough? The tests that are currently failing seemingly have nothing to do with the changes I've made.

File: src/server.ts [hunk 1]
```diff
@@ -406,10 +406,15 @@ export class Server extends EventEmitter {
         if (authResult instanceof Error) {
           return this._sendError({
             Code: {
-              Value: 'SOAP-ENV:Server',
-              Subcode: { value: 'InternalServerError' },
+              Value: 'soap:Receiver',
```
Comment (jsdevel) [inline src/server.ts]: so it's been like this for a while, why does this need to be changed?

Comment (sigitas-stumbras) [reply src/server.ts]: This was not according to soap envelope 1.2 schema: https://www.w3.org/2003/05/soap-envelope/
valid values for Fault->Code->Value are
```
<xs:enumeration value="tns:DataEncodingUnknown"/>
<xs:enumeration value="tns:MustUnderstand"/>
<xs:enumeration value="tns:Receiver"/>
<xs:enumeration value="tns:Sender"/>
<xs:enumeration value="tns:VersionMismatch"/>
```

and the namespace tag "soap" for "https://www.w3.org/2003/05/soap-envelope/" is being declared in src/server.ts line: 589-596 
tag "SOAP-ENV is not being declared anywhere

thus the **soap:Server**

fully schema compliant Saop Envelope 1.2 Fault reponse is like this:
```
<env:Envelope xmlns:env="http://www.w3.org/2003/05/soap-envelope"
              xmlns:m="http://www.example.org/timeouts"
              xmlns:xml="http://www.w3.org/XML/1998/namespace">
 <env:Body>
  <env:Fault>
   <env:Code>
     <env:Value>env:Sender</env:Value>
     <env:Subcode>
      <env:Value>m:MessageTimeout</env:Value>
     </env:Subcode>
   </env:Code>
   <env:Reason>
     <env:Text xml:lang="en">Sender Timeout</env:Text>
   </env:Reason>
   <env:Detail>
     <m:MaxTime>P5M</m:MaxTime>
   </env:Detail>    
  </env:Fault>
 </env:Body>
</env:Envelope>
```

Comment (sigitas-stumbras) [reply src/server.ts]: What would be nice to have is the possibility to switch to SOAP 1.1 envelope Fault response. It has a different structure. Maybe use the existing forceSoap12Headers option for this? or maybe create a new option?

**Comment by jsdevel** [comment]
ok.  once the build passes, i'll merge it.  can you look into that?

**Comment by sigitas-stumbras** [comment]
All green & commits squashed. All done now?

File: test/request-no-envelope-body-test.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ describe('No envelope and body elements', function() {
         if(err){
           throw err;
         }
-        assert.equal(body.indexOf('Failed to parse the SOAP Message body') !== -1, true);
+        assert.equal(body.indexOf('InternalServerError') !== -1, true);
```
Comment (jsdevel) [inline test/request-no-envelope-body-test.js]: it would be nice to have a more descriptive error here

**Comment by nraynaud** [comment]
I'd like to bring your attention to this code branch https://github.com/vpulim/node-soap/blob/master/src/wsdl/index.ts#L504 where the fault management is using the wrong case in the tag names. Is it linked to the same SOAP version issue? I am unfamiliar with SOAP and my use case is windows wsman.

**Comment by sacru2red** [comment]
The PR was closed like this without merging. Can I reopen it and process it? @jsdevel

**Comment by w666** [comment]
I think @sigitas-stumbras deleted forked repo.
I am happy to review it if you re-open the PR @sacru2red.

**Pull request closed without merging**